pkg/scheduler/core/generic_scheduler.go

```diff
@@ -146,6 +152,24 @@ func (g *genericScheduler) snapshot() error {
 // If it succeeds, it will return the name of the node.
 // If it fails, it will return a FitError error with reasons.
 func (g *genericScheduler) Schedule(ctx context.Context, prof *profile.Profile, state *framework.CycleState, pod *v1.Pod) (result ScheduleResult, err error) {
+	annotation, ok := pod.Annotations["trace.kubernetes.io/context"]
+	if ok {
+		propagator := trace.DefaultHTTPPropagator()
+		reqHeader := &http.Header{}
+		decodedContextBytes, err := base64.StdEncoding.DecodeString(annotation)
+		if err != nil {
+			return result, err
+		}
+		err = json.Unmarshal(decodedContextBytes, reqHeader)
+		if err != nil {
+			return result, err
+		}
+		ctx = propagator.Extract(ctx, reqHeader)
+	}
+	ctx, span := global.TraceProvider().Tracer("new-scheduler").Start(ctx, fmt.Sprintf("genericScheduler: Schedule '%s'", pod.Name))
```

Comment:
as `pod.Name` list may be large, was it intentional to have a span name include the `pod.Name`? It is advisable to keep the cardinality of span names low and make sure that span name represents the proper grouping of spans by the latency. if it's intended, it's ok.

Reply:
In my test runs I found it a lot easier to "browse" the traces for the pod I was looking for when the pod name was in the span name. However I realize this could also be easily accomplished with annotations and see your point.